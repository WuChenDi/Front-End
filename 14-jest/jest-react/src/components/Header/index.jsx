import { useState } from "react";
import "./index.css";

const Header = (props) => {
  const [inputValue, setInputValue] = useState("");
  const { add } = props;

  return (
    <div className="header">
      <span>ToDoList</span>
      <input
        className="header-input"
        data-test="input"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
        onKeyUp={(e) => {
          if (e.keyCode === 13 && inputValue) {
            add?.(inputValue);
            setInputValue('')
          }
        }}
        placeholder="add ToDoItem"
      />
    </div>
  );
};

export default Header;
