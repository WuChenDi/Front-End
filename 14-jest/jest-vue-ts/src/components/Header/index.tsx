import { defineComponent, ref } from "vue";
import { HeaderPropsDefine } from "@/utils/types";
import "./index.scss";

export default defineComponent({
  name: "Header",
  props: HeaderPropsDefine,
  setup(props) {
    const inputValueRef = ref<string>("");

    const addToDoItem = () => {
      props?.add(inputValueRef.value);
      inputValueRef.value = "";
    };

    return () => {
      const inputValue = inputValueRef.value;

      return (
        <div class="header">
          <span>ToDoList</span>
          <input
            class="header-input"
            data-test="input"
            v-model={inputValue}
            onKeyup={(e: any) => {
              if (e.keyCode === 13 && inputValue) {
                addToDoItem();
              }
            }}
            placeholder="add ToDoItem"
          />
        </div>
      );
    };
  },
});
