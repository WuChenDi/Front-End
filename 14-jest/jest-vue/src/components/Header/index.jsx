// @ts-nocheck
import "./index.scss";

const Header = {
  name: "Header",
  data() {
    return {
      inputValue: "",
    };
  },
  methods: {
    addToDoItem() {
      if (this.inputValue) {
        this.$emit("add", this.inputValue);
        this.inputValue = "";
      }
    },
  },
  render() {
    return (
      <div class="header">
        <div class="header-content">
          ToDoList:
          <input
            class="header-input"
            data-test="input"
            v-model={this.inputValue}
            vOn:keyup_enter={this.addToDoItem}
            placeholder="add ToDoItem"
          />
        </div>
      </div>
    );
  },
};

export default Header;
