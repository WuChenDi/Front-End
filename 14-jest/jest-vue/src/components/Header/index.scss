.header {
  line-height: 60px;
  background-color: #333;
}
.header-content {
  width: 600px;
  margin: 0 auto;
  color: #fff;
  font-size: 24px;
}
.header-input {
  float: right;
  width: 360px;
  margin-top: 16px;
  line-height: 24px;
  outline: none;
  color: #333;
  text-indent: 10px;
}
