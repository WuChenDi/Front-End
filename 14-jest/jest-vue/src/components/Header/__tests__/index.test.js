import { shallowMount } from "@vue/test-utils";
import Header from "../index.jsx";

describe("Header.vue", () => {
  it("Header 样式发生改变，做提示", () => {
    const wrapper = shallowMount(Header);
    expect(wrapper).toMatchSnapshot();
  });

  it("Header 包含 input 框", () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    expect(input.exists()).toBe(true);
  });

  it("Header 中 input 框初始值为空", () => {
    const wrapper = shallowMount(Header);
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("");
  });

  it("Header 中 input 框值发生变化，数据应该跟着变", () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue("hello");
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("hello");
  });

  it("Header 中 input 框输入回车，无内容时，无反应", () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue("");
    input.trigger("keyup.enter");
    // expect(wrapper.emitted.add).toBeTruthy();
    expect(wrapper.emitted().add).toBeFalsy();
  });

  it("Header 中 input 框输入回车，有内容时，向外触发事件", () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue("hello");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeTruthy();
  });

  it("Header 中 input 框输入回车，有内容时，向外触发事件，同时清空 inputValue", () => {
    const wrapper = shallowMount(Header);
    const input = wrapper.find('[data-test="input"]');
    input.setValue("hello");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeTruthy();
    expect(wrapper.vm.$data.inputValue).toBe("");
  });
});
