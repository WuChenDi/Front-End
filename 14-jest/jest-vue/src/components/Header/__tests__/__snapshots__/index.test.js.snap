// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue Header 样式发生改变，做提示 1`] = `<div class="header"><span>ToDoList</span><input data-test="input" placeholder="add ToDoItem" class="header-input"></div>`;
